python/plugins/db_manager/db_plugins/gpkg/connector.py

```diff
@@ -39,6 +39,8 @@
 import sqlite3
 
 from osgeo import gdal, ogr, osr
+gdal.UseExceptions()
+ogr.UseExceptions()
```

Comment:
Should this be globally enabled instead of inside this plugin?
The effect is global, so doing this here might lead to hard to track issues for other plugins.

Reply:
good point. Would ``python/__init__.py`` be the right place to do it ? (A potential downside could be a slight downside of the QGIS startup)

Reply:
I wouldn't know a better place. Are we talking about a noticeable slowdown/cost?

Reply:
> Are we talking about a noticeable slowdown/cost?

150 ms on my Linux system after "echo 3 > /proc/sys/vm/drop_caches" . 100 ms on hot runs

Reply:
Can we revisit this choice? I spend quite some time last cycle ensuring that qgis_process startup time was as small as possible, and I'd rather not introduce this startup cost if we can possibly avoid it.

Could we possibly move this code to QgisApp::loadPythonSupport() instead? I realise it's not a perfect solution, because standalone scripts would exhibit different behavior, but I think it's an acceptable trade off.

Reply:
> Could we possibly move this code to QgisApp::loadPythonSupport() instead?

hum, well ``python/__init__.py`` is called indirectly from loadPythonSupport():
```
#40 0x00007fbdacc304f9 in PyRun_StringFlags () at /lib/x86_64-linux-gnu/libpython3.10.so.1.0
#41 0x00007fbdcf396217 in QgsPythonUtilsImpl::runStringUnsafe(QString const&, bool) (this=0x5612fe897470, command=..., single=true) at /home/even/qgis/qgis/src/python/qgspythonutilsimpl.cpp:388
#42 0x00007fbdcf396461 in QgsPythonUtilsImpl::runString(QString const&, QString, bool) (this=0x5612fe897470, command=..., msgOnError=..., single=true)
    at /home/even/qgis/qgis/src/python/qgspythonutilsimpl.cpp:405
#43 0x00007fbdcf393a1e in QgsPythonUtilsImpl::checkSystemImports() (this=0x5612fe897470) at /home/even/qgis/qgis/src/python/qgspythonutilsimpl.cpp:147
#44 0x00007fbdcf3951d5 in QgsPythonUtilsImpl::initPython(QgisInterface*, bool, QString const&) (this=0x5612fe897470, interface=0x5612fe7ff160, installErrorHook=true, faultHandlerLogPath=...)
    at /home/even/qgis/qgis/src/python/qgspythonutilsimpl.cpp:255
#45 0x00007fbe52afef93 in QgisApp::loadPythonSupport() (this=0x5612fd17f870) at /home/even/qgis/qgis/src/app/qgisapp.cpp:12516
```
Not sure to understand what you mean. Or do you mean to inline the few lines added in __init__.py in qgspythonutilsimpl.cpp and execute them with runString() ? But I don't see how that would change performance at all...

Reply:
@rouault I'm trying to avoid the cost getting incurred here: https://github.com/qgis/QGIS/blob/master/src/process/qgsprocess.cpp#L237

(ie when we're not using qgisapp)

Reply:
@nyalldawson Got it. I've removed the additions in ``python\__init__.py`` and added https://github.com/qgis/QGIS/pull/57477/commits/d2549796ea18fb06bb8b7668cc42bdd1ab5d7aee